jmix-reports/reports/src/main/java/io/jmix/reports/libintegration/JmixObjectToStringConverter.java

```diff
@@ -66,12 +68,36 @@ public Object convertFromString(Class parameterClass, @Nullable String paramValu
         } else if (String.class.isAssignableFrom(parameterClass)) {
             return paramValueStr;
         } else if (Entity.class.isAssignableFrom(parameterClass)) {
-            UUID id = UUID.fromString(paramValueStr);
-            Optional entityOpt = dataManager.load(parameterClass)
-                    .id(id)
-                    .fetchPlan(FetchPlan.BASE)
-                    .optional();
-            return entityOpt.isPresent() ? entityOpt.get() : null;
+            MetaProperty idProperty = metadataTools.getPrimaryKeyProperty(parameterClass);
+            if (idProperty == null) {
+                return null;
+            }
+
+            Object idValue;
+
+            if (idProperty.getRange().isDatatype()) {
+                try {
+                    idValue = idProperty.getRange().asDatatype().parse(paramValueStr);
+                } catch (ParseException e) {
+                    throw new ReportingException(
+                            String.format("Couldn't read id from [%s] with value [%s] and datatype [%s].",
+                                    parameterClass.getSimpleName(), paramValueStr, idProperty.getRange().asDatatype()));
+                }
+
+                if (idValue != null) {
+                    return dataManager.load(parameterClass)
+                            .id(Id.of(idValue, parameterClass))
+                            .fetchPlan(FetchPlan.BASE)
+                            .optional()
+                            .orElse(null);
+                }
+            } else if (idProperty.getRange().isClass()) {
+                throw new ReportingException(
+                        String.format("Unsupported composite id from [%s] with value [%s]",
```

Comment:
Let's rephrase: "composite id from" -> "composite primary key in"

Reply:
Fixed